Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -26,13 +26,12 @@ class PreplannedMapModel: ObservableObject, Identifiable {
     
     init(preplannedMapArea: PreplannedMapAreaProtocol) {
         self.preplannedMapArea = preplannedMapArea
-        
-        // Kick off a load of the map area.
-        Task.detached { await self.load() }
     }
     
     /// Loads the preplanned map area and updates the status.
-    private func load() async {
+    func load() async {
+        if case .packaged = status { return }
+        if case .downloaded = status { return }
```

Comment:
is there a reason to go with `if` instead of `guard` here? `guard` is tailor made for early exit of a function

Reply:
I don't know what is the syntax of using `guard` here. It cannot do something like `guard case .packaged != status else { return }` because guard-case or if-case behave somewhat like guard-let and if-let, which doesn't allow `!=`. Also, since the `Status` enum is not `Equatable`, we cannot simply use `==` to compare the enum cases.

Alternatively as I mentioned, we can use switch-case here such as`

```swift
switch status {
    case .packaged, .downloaded: return
    default: break
}
```

if you prefer that, I can make the change.

Reply:
I would do something like this:

```swift

    func foo(status: Foo) {
        guard status.needsToBeLoaded else { return }
        /// ...
    }

enum Foo {
    case loaded
    case loading
    case packaged
    case failure(any Error)
    
    var needsToBeLoaded: Bool {
        switch self {
        case .loaded:
            false
        case .loading:
            false
        case .packaged:
            false
        case .failure(let error):
            true
        }
    }
}

```